### PR TITLE
Katie's JS-Adagrams

### PR DESCRIPTION
**Pull request by katietj**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | I was able to use my Ruby knowledge to say, 'ok, I know how to to this in Ruby...can I just do the same thing in js?" and I would try it. Much of what you can do in Ruby, you can replicate in js...just with different syntax. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | No, I used the same methods....google and stack overflow were helpful. The MDN guide was also useful. 
What was a challenge you were able to overcome on this assignment? | I had to fight a little bit of frustration in this assignment when doing something that seemed easy in Ruby was tougher in JS due to me not knowing the syntax super well. 
What is your favorite thing about learning a new programming language? | I like that you don't have to start from square one, I'm seeing now that so much of js is like Ruby and it's going be a much quicker learning experience this time around. 
What is your least favorite thing about learning a new programming language? | All the little roadblocks that you hit trying to do basic tasks in new language. It slows down your progress and can feel frustrating. 
Do you have any recommendations on how we could improve this project for the next cohort? | I looked at the instructor's solution for the original Adagrams and that was helpful because my Ruby Adagrams was not great....maybe remind the cohort that this is an option. I think this helped me because I could focus on the ruby to js translation and wasn't trying to refactor the ruby code at the same time.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,124 @@
...
+       drawnLetters.push(randomLetter);
+    }
+
+  return   drawnLetters
+
   },
+  usesAvailableLetters(input, lettersInHand) {
+    let valid = true;
+    let inputCharacters = input.toUpperCase().split('');
+    let newHand = lettersInHand.map(letter => letter);
```
Comment (Hamled) [inline src/adagrams.js:46]: Nice code to copy this array! We could also have done `let newHand = [...lettersInHand];`.

**Comment by Hamled** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | :white_check_mark:
Small commits with meaningful commit messages | :white_check_mark:
**Code Requirements** |
`drawLetters` method | :white_check_mark:
Uses appropriate data structure to store the letter distribution | :white_check_mark:
All tests for `drawLetters` pass | :white_check_mark:
`usesAvailableLetters` method | :white_check_mark:
All tests for `usesAvailableLetters` pass | :white_check_mark:
`scoreWord` method | :white_check_mark:
Uses appropriate data structure to store the letter scores | :white_check_mark:
All tests for `scoreWord` pass | :white_check_mark:
`highestScoreFrom` method | :white_check_mark:<br>The indentation style for the `forEach` loop code in this function is pretty gnarly...
Appropriately handles edge cases for tie-breaking logic | :white_check_mark:
All tests for `highestScoreFrom` pass | :white_check_mark:
**Overall** | Looks great. This code is very clean in general (except for the code style for the `highestScoreFrom` function). Good work!
